composer/trainer/trainer.py

```diff
@@ -274,13 +289,15 @@ def __init__(
         self.state.schedulers = ComposedScheduler(schedulers=schedulers)
 
         self.checkpointer = None
-        if checkpoint_folder and checkpoint_interval and checkpoint_interval_unit:
+        # TODO: get checkpointing working with DeepSpeed.
```

Comment:
can you raise a `NotImplementedError` here if someone tries to use checkpointing and deepspeed?